src/Nethermind/Nethermind.Consensus.Ethash/DifficultyCalculator.cs

```diff
@@ -59,26 +59,21 @@ public DifficultyCalculator(ISpecProvider specProvider)
             {
                 return BigInteger.Max((parentHasUncles ? 2 : BigInteger.One) - BigInteger.Divide(currentTimestamp - parentTimestamp, 9), -99);
             }
-            
-            if(spec.IsEip2Enabled)
-            {
-                return BigInteger.Max(BigInteger.One - BigInteger.Divide(currentTimestamp - parentTimestamp, 10), -99);    
-            }
-            
-            if (spec.IsTimeAdjustmentPostOlympic)
+
+            if (spec.IsEip2Enabled)
             {
-                return currentTimestamp < parentTimestamp + 13 ? BigInteger.One : BigInteger.MinusOne;                
+                return BigInteger.Max(BigInteger.One - BigInteger.Divide(currentTimestamp - parentTimestamp, 10), -99);
             }
-            
-            return currentTimestamp < parentTimestamp + 7 ? BigInteger.One : BigInteger.MinusOne;
+
+            return currentTimestamp < parentTimestamp + (spec.IsTimeAdjustmentPostOlympic ? 13 : 7) ? BigInteger.One : BigInteger.MinusOne;
         }
 
         private BigInteger TimeBomb(IReleaseSpec spec, long blockNumber)
         {
-            blockNumber = blockNumber - spec.DifficultyBombDelay;
+            blockNumber -= spec.DifficultyBombDelay;
 
             // Note: block 200000 is when the difficulty bomb was introduced but we did not spec it in any release info, just hardcoded it
-            return blockNumber < 200000 ? UInt256.Zero : BigInteger.Pow(2, (int)(BigInteger.Divide(blockNumber, 100000) - 2));
+            return blockNumber < 200000 ? BigInteger.Zero : BigInteger.Pow(2, (int)(BigInteger.Divide(blockNumber, 100000) - 2));
```

Comment:
You are introducing an unnecessary cast of BigInteger.Zero to UInt256.Zero

Reply:
It's the exactly the opposite.
Function's return type is BigInteger: previous code caused the double implicit casting. Not this one.